SecurityPkg/VariableAuthenticated/SecureBootConfigDxe/SecureBootConfigImpl.c

```diff
@@ -2539,16 +2540,22 @@ UpdateDeletePage (
   EFI_SIGNATURE_LIST  *CertList;
   EFI_SIGNATURE_DATA  *Cert;
   UINT32              ItemDataSize;
-  CHAR16              *GuidStr;
+  CHAR16              *CertificateInfoStr;
+  CHAR8               *CertificateInfoStr8;

```

Comment:
If the probles is caused by lack of heap space, try allocating these on stack:

```suggestion
  CHAR16              CertificateInfoStr[100];
  CHAR8               CertificateInfoStr8[100];
```

Note that some of my other comments don't apply if you choose to do so.

Reply:
CertificateInfoStr8 needs only 50 bytes right?

Reply:
That would be the case there. Note that the code has changed now according to your comments and they all are outdated now.